qubes/storage/__init__.py

```diff
@@ -339,6 +381,43 @@ def size(self, size):
         # pylint: disable=attribute-defined-outside-init
         self._size = int(size)
 
+    # pylint: disable=no-self-use
+    def volatile_volume_path(self, qube_name, device_name):
+        '''Find the name of the encrypted volatile volume'''
+        # We need to ensure we don’t collide with any name used by LVM or LUKS,
+        # and that different qubes have different encrypted volume names.
+        # LUKS volumes have a name starting with `luks-` followed by a UUID.
+        # LVM volumes always have at most one dash that is not doubled.
+        # And there is a one-to-one relationship between escaped and original
+        # names: replace `_d` with `-`, then replace `_u` with `_`.
+        # So we are in the clear here.
+        escaped_qube_name = qube_name.replace('_', '_u').replace('-', '_d')
+        escaped_vol_name = \
+            'vm-volatile-' + escaped_qube_name + '-crypt@' + device_name
+        assert escaped_vol_name[12:-7-len(device_name)] == escaped_qube_name
+        assert escaped_qube_name.replace('_d', '-').replace('_u', '_') == \
+            qube_name
+        return escaped_vol_name
+
+    def make_encrypted_device(self, device, qube_name):
+        ''' Takes :py:class:`BlockDevice` and returns its encrypted version for
+            serialization in the libvirt XML template as <disk>.  The qube name
+            is available to help construct the device path.
+        '''
+        assert device.domain is None, "Volatile volume must be in dom0"
+        assert device.script is None, \
+            "Block device scripts for ephemeral volumes not supported"
+        assert device.devtype == 'disk'
+        assert device.rw, 'Encrypting read-only volumes makes no sense'
+        path = self.volatile_volume_path(qube_name, device.name)
+        return qubes.storage.BlockDevice(
+            path='/dev/mapper/' + path,
```

Comment:
In LVM pool you already return the full path from `volatile_volume_path`. And given the `path` in the function name, I'd expect it this way.